pkg/resourcemanager/mysql/mysqlhelper.go

```diff
@@ -168,7 +213,53 @@ func GrantUserRoles(ctx context.Context, user string, database string, roles []s
 	return nil
 }
 
-func addRoles(ctx context.Context, db *sql.DB, database string, user string, roles map[string]struct{}) error {
+// EnsureUserDatabaseRoles revokes and grants database roles as needed
+// so they match the ones passed in. It returns warning messages if
```

Comment:
I'm not totally clear how the comment: 
> It returns warning messages if some databases don't exist

is correct?

1. Is `ExtractUserDatabaseRoles` succeeding (but with 0 roles) if the DB doesn't exist? It's not clear to me if that's an error case or an "empty return" case.
2. Other errors when adding/removing roles will also just get reported as warnings right? As a matter of fact based on my read of this function and its caller it seems like even if we cannot assign some roles to some DBs we will still say `Provisioned = true` and `State = "Succeeded"` - that seems probably not correct?

I can sorta see the reasoning around bypassing DB not exists (as maybe next reconcile loop it will), but wonder if we need to be more explicit about checking for errors that aren't that one. Even for that error it might make sense to leave this resource in a "provisioned=false" state as it's not done with all that was asked?

Reply:
This makes a lot of sense - after our discussion I'll accumulate the errors for different databases into one error from the Ensure function (even though we still apply the roles for the later databases on an error) and have the user still be provisioned=false in the reconciler.